src/postgres/commands/checkAuthentication.ts

```diff
@@ -20,7 +20,7 @@ export async function checkAuthentication(context: IActionContext, treeItem: Pos
 
             if (parsedError.errorType === invalidCredentialsErrorType) {
                 await enterPostgresCredentials(context, treeItem.parent);
-            } else if (parsedError.errorType === firewallNotConfiguredErrorType) {
+            } else if (treeItem.parent.resourceGroup && parsedError.errorType === firewallNotConfiguredErrorType) {
```

Comment:
It's not automatically obvious why `configurePostgresFirewall` needs a resource group, but `enterPostgresCredentials` doesn't. Either change the code or add a comment (in the code, not here) why you're doing what you're doing